go/enclave/components/batch_executor.go

```diff
@@ -207,6 +209,15 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext) (*Co
 	}
 
 	executor.populateHeader(&copyBatch, allReceipts(txReceipts, ccReceipts))
+	if failForEmptyBatch &&
+		len(txReceipts) == 0 &&
+		len(ccReceipts) == 0 &&
+		len(transactionsToProcess) == 0 &&
+		len(crossChainTransactions) == 0 &&
+		len(messages) == 0 &&
+		len(transfers) == 0 {
+		return nil, ErrNoTransactionsToProcess
```

Comment:
The AI bunny is right to be concerned I think. I don't like adding the lint ignore above and this approach does feel messy. But by bailing out within this method it's easy to be sure there was nothing important going to be in the batch and it also avoid committing anything to the stateDB cache.

Any ideas for how to do this more cleanly?

Reply:
I think it's ok